src/test/java/com/google/roomies/servlets/ListingsServletTest.java

```diff
@@ -76,6 +88,53 @@ public void setUp() throws Exception {
 
     DatabaseFactory.setDatabaseForTest(db);
     setRequestParameters();
+    defineDatabaseMockBehavior();
+
+    stringWriter = new StringWriter();
+    when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+ 
+  @Test
+  public void testGet_returnsSingleListing() throws Exception {
+    Map<String, Object> listingData = mapOfListingDataForGetTests(request);
+    when(queryDocumentMock.getData()).thenReturn(listingData);
+ 
+    listingsServlet.doGet(request, response);
+    String expectedWriterOutput =  "[{\"documentId\":{\"value\":\"documentID\"}," + 
+      "\"timestamp\":{\"value\":{\"seconds\":1474156800,\"nanos\":0}},\"title\":\"Test title\"," +
+      "\"description\":\"Test description\",\"startDate\":\"Jul 10, 2020, 12:00:00 AM\"," + 
+      "\"endDate\":\"Jul 10, 2020, 12:00:00 AM\",\"leaseType\":\"YEAR_LONG\",\"numRooms\":2," +
+      "\"numBathrooms\":2,\"numShared\":2,\"numSingles\":2,\"sharedPrice\":\"USD 100\","+
+      "\"singlePrice\":\"USD 0\",\"listingPrice\":\"USD 100\"}]";
+
+    verify(db, Mockito.times(1)).getAllDocumentsInCollection(LISTING_COLLECTION_NAME);
+    verify(queryDocumentMock, Mockito.times(1)).getData();
```

Comment:
Are testing these mock interactions explicitly needed? Can the return value be correct without them? 

If you have a reason you /do/ need to care about verifying the mock interaction, then that would constitute a separate behavior and a separate unit test.